realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -154,11 +154,15 @@ public void addChangeListener(RealmChangeListener listener) {
      * Removes the specified change listener.
      *
      * @param listener the change listener to be removed.
+     * @throws IllegalStateException if you try to remove a listener from a non-Looper Thread.
      * @see io.realm.RealmChangeListener
      * @see #addChangeListener(RealmChangeListener)
      */
     public void removeChangeListener(RealmChangeListener listener) {
         checkIfValid();
+        if (handler == null) {
```

Comment:
I am still not sure this is really needed. At worst it would be a no-op since you aren't allowed to add it in the first place as far as I can tell?